test/logical-search.test.js

```diff
@@ -240,3 +240,25 @@ describe('Logical search with dotted keys', () => {
     expect(result.length).toBe(1)
   })
 })
+
+describe('Searching using logical OR with same query across fields', () => {
+  const options = { keys: ['title', 'author.lastName'] }
+  const fuse = new Fuse(Books, options)
+  let result
+  beforeEach(() => {
+    const query = {
+      $or: [
+        { title: 'wood' },
+        { 'author.lastName': 'wood' }
+      ]
+    }
+
+    result = fuse.search(query)
+  })
+
+  describe('When searching for the term "wood"', () => {
+    test('we get the top three results all with an exact match from their author.lastName', () => {
+      expect(idx(result.slice(0,3)).sort()).toMatchObject([3,4,5])
```

Comment:
Shouldn't the order be 4,3,5? I am imagining it would have to match the following:

```js
{
    item: {
    title: 'The Code of the Wooster',
    author: { firstName: 'P.D', lastName: 'Woodhouse' }
    },
    refIndex: 4
},
{
    item: {
    title: 'Right Ho Jeeves',
    author: { firstName: 'P.D', lastName: 'Woodhouse' }
    },
    refIndex: 3
},
{
    item: {
    title: 'Thank You Jeeves',
    author: { firstName: 'P.D', lastName: 'Woodhouse' }
    },
    refIndex: 5
}
```

But, I think the reason the order is 3,4,5 is because of line 185: `res = [...result]`. Did you mean to do `res.push(...result)`?

Reply:
@krisk I suppose that would be a design decision. Do we want to include all matches from all fields in the OR which have a match, or do we want to only score based on the best match out of all the fields scanned by the OR.

Happy to switch to `res.push(...result)` as I feel that would be the better design decision (include score from all matches in logical OR).

Reply:
I've wavered on this for a while. I do think including scores for all matches is ideal.